### ISSUE TITLE
display problems

### ISSUE THREAD

**Issue by Johnephen** [penk/ruha.camera#11]
I took a 2.2 "TFT display as recommended and connected it, but a problem emerged.
Display Link (https://www.amazon.de/gp/product/B01E0MGZU2/ref=ppx_yo_dt_b_asin_title_o07_s00?ie=UTF8&psc=1)

In a blog it was shown how it can be connected to the Raspberry Pi and in the linked Github project this was also discussed again. But I have a problem that wasn't described there. After starting the script, the image can only be seen in the upper right part of the display, greatly reduced. Unfortunately, the autostart script does not work properly either. I have to run the script to add it to the autostart after every reboot. In addition, the displayed image flickers as if the frame rate is incorrect.
The Blog(in German)(https://blog.gc2.at/post/spi-tft-lcd-kernel5/)
the GitHubProjekt(english)(https://github.com/juj/fbcp-ili9341)

I very much hope that a solution can be found and that everything works out. Many thanks in advance.

**Comment by penk** [comment]
Hi @Johnephen, thank you for the issue report.

It seems like in your current implementation, you're using a different kernel driver (mentioned in the blog) and fbcp command to mirror main output to your display. There are many variables here (and many options) so let me try to provide my way of thinking for your reference: 

- For the resolution (image only in upper right part) issue, I would check if `gpu_mem=256` is added in `/boot/config.txt`. It's essential for the camera to take the full resolution. 
- For the not-auto-start script issue, I'd recommend adding a `.desktop` file under `/etx/xdg/autostart` folder. You can find more information in this link: https://help.gnome.org/admin/system-admin-guide/stable/autostart-applications.html.en
(And of course you'll need to change the `Exec=` line to the path of your script) 
- The flicking and frame rate issue seems to be related to the fbcp command itself, I'd check if correct build flags are used in compiling fbcp

And finally, if it still fails you, please try using the original Raspberry Pi OS overlay driver like I did and see if it works better for you. We had a ticket on that here: 
https://github.com/penk/ruha.camera/issues/11 

Good luck!

**Comment by Johnephen** [comment]
Thank you for this so quick response. I kept looking around for display solutions and a video that actually solved all of my display issues. The autostart also works and the camera script also works. So when I start the Raspberry Pi, I also have a playback of the camera. But I'm still a little dissatisfied with the refresh rate of the playback itself, since that of the display should actually be ok. But I'll look at that again later, whether it's really on the display or on the playback. But thanks again for the answer.

Video I used: https://www.youtube.com/watch?v=uz4sa24C4gk&list=LL&index=4&t=1015s
